src/map/Map.kt

```diff
@@ -0,0 +1,41 @@
+package map
+
+import java.util.ArrayList
+class Map(private val nodes: Array<Array<Node>>) {
+    companion object {
```

Comment:
That is a lot of boiler-plate...

Also, why static again? Having multiple different maps per process kinda makes sense, eg if you choose to run multiple games in parallel. Or have different maps. Etc.